Updates to access control documentation (#7730)
keystonejs/keystone: docs/pages/docs/apis/access-control.mdx

## List Access Control

Keystone provides support for access control and filtering on a per-list basis.
By default, all lists and fields are public without additional configuration.

There are three types of access control that can be configured on the `access` object:

> - [`operation`](#operation), functions which provide the `context` and `session` to help decide if a request should be granted access to the list; this happens prior to any GraphQL query being executed.

Reviewer: ```suggestion
- [`operation`](#operation), functions that provide the `context` and `session` to help decide if a request should be granted access to the list; this happens prior to any GraphQL query being executed.
```

- [`filter`](#filter), functions which return a [filter](./filters) that is applied to database queries on this list.

> - [`item`](#item-mutations-only), functions which provides the `context` and `session` to determine if a request can execute the respective mutative action.

Reviewer: ```suggestion
- [`item`](#item-mutations-only), functions that provide the `context` and `session` to determine if a request can execute the respective mutative action.
```

});
```

!> Filter based access control cannot be used for `create` operations.
If you want to limit `create` operations, use either `access.operation.create` or `access.item.create`.

?> Filter based access control can impact the performance of your database queries.

### Item (mutations only)

> Item-level access control lets you control which mutative operations can be applied to a list item.

Reviewer 2: > mutative

Would _mutable_ be more apt here?

Reviewer: Or just
```
Item-level access control lets you control which operations can be applied to a list item
```
OR
```
Item-level access control lets you control which mutations can be applied to a list item
```
My preference is 1...

Individual functions can be provided for each of the operations, with each function receiving:

> - the **input data** of the mutation (for `create` and `update` operations), and or

Reviewer: ```suggestion
- the **input data** of the mutation (for `create` and `update` operations), and/or
```
